src/compiler/checker.ts

```diff
@@ -2766,7 +2766,9 @@ namespace ts {
         }
 
         function getBaseTypeNodeOfClass(type: InterfaceType): ExpressionWithTypeArguments {
-            return getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>type.symbol.valueDeclaration);
+            let classDeclaration =
+                getDeclarationOfKind(type.symbol, SyntaxKind.ClassDeclaration) || type.symbol.valueDeclaration;

```

Comment:
There's nothing that says `valueDeclaration` has to be the first declaration. In fact, it's not even all that clear what "first one" means and ultimately might depend on the order in which files were specified on the command line. But was is clear is that in cases where a namespace is merged with a function, class, or enum, we always want to treat the combined entity as the non-namespace kind, and the purpose of `valueDeclaration` is (or should be) to be a reference to that "primary" declaration.